### PR TITLE
Add BigInteger tutorial

### PR DESCRIPTION
**Pull request by cjn10**
This pull request adds a small tutorial about the `BigInteger` class and how to perform resource estimation of the order-finding quantum routine. It also changes the `__call__` method of the `BigInteger` to return a Python integer in non-jasp mode.

### CODE REVIEW

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1dd4e9a8",
+   "metadata": {},
+   "source": [
+    "Here, we perform the same computation using Python and Jasp and get different results.\n",
+    "Additionally, we are not alerted to the problem since Jasp does not check bound violations on all internal computations.\n",
+    "\n",
+    "This issue is known as _Integer overflow_ and is a common pitfall in many low-level programming languages.\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: Yes, but also in numpy, jax.numpy. Maybe slightly repharse. We don't want to compare to low-level languages :).

File: documentation/source/_static/order_finding_qre.svg
Comment (renezander90) [inline documentation/source/_static/order_finding_qre.svg:1]: Please add this plot to the tutorial with some explanation. Axes labels could be slightly larger. This is qrisp blue:  '#20306f'

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    "It is a direct consequence of setting the size of the integer to static value, for example, 32- or 64-bit, and many other basic data types are handled the same way.\n",
+    "Doing this allows for well-optimized operations and avoids additional overhead introduced by managing a dynamically-sized data container.\n",
+    "In fact, we are not able to circumvent this restriction in general because of the limitations of Jasp.\n",
+    "Still, to allow us to work with integers that are larger than the native 32- or 64-bit types, we provide the `BigInteger` class.\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: Use [BigInteger] (../../reference/Jasp/Scalable%20Integer%20Type.rst)

Without the gap between ] and (

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    "\n",
+    "The idea of the `BigInteger` class is easily explained: represent very large integers that do not fit into the default sizes.\n",
+    "Internally, the `BigInteger` uses an array of 32-bit integers to represent larger values.\n",
+    "As required by Jasp, this array possesses a static size.\n",
+    "Now, we will see how to work with this integer.\n",
+    "\n",
+    "## Working classically with `BigInteger`\n",
+    "\n",
+    "To create an instance of the `BigInteger` class, we can make use of the `create_static` or `create_dynamic` function.\n",
+    "Where `create_static` is a Python function that is *not jasp-compatible*, the function `create_dynamic` is written in terms of jasp primitives.\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: This can be confusing. create_static is used with Jasp, but the argument has to be static.

And, I would write Jax primates instead of Jasp primitives.

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "b3ef0837",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@jaspify\n",
+    "def create_bi():\n",
+    "    return BigInteger.create_dynamic(1, 64) # Works because the value is one\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: Works because the value 1 fits into 64 bits. Maybe use something larger than 1.

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    "print(\"bi_large:\", bi_large())\n",
+    "print(\"Correct value:\", bi_large() == 2**1200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cec6b446",
+   "metadata": {},
+   "source": [
+    "In Python-mode, that is, outside any jasp function, the method returns a Python integer of the value stored inside the `BigInteger` instance.\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: jasp -> Jasp (here and elsewhere)

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d678d75d",
+   "metadata": {},
+   "source": [
+    "The function `find_order` provides a jasp-compatible implementation of order-finding that makes use of the BigIntger class.\n",
+    "Note that we use the function `measure_to_big_integer` instead of `measure` to return the measured value of the quantum variable as a big integer.\n",
+    "This use case shows the importance of the `BigInteger` class, since a native jasp integer is limited to 64 bits.\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: Jax integer

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+    "Only minimal modifications are required to make the implementation compatible with the new integer type.\n",
+    "\n",
+    "Of course, we cannot run this function for, say, 2048-bit integers on a classical computer.\n",
+    "Still, we are able to perform resource estimation using this implementation."
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: Here, we run resource estimation of 128-bit integers.

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+    "ops_count = find_order_count_ops(a, N)\n",
+    "print(ops_count)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f4539fcd",
+   "metadata": {},
+   "source": [
+    "Using the `count_ops` annotation, we can obtain a dictionary containing the number and type of quantum gates that are used in the implementation."
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: Please add the plot here.

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 9]
```diff
@@ -0,0 +1,360 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "403ca162",
+   "metadata": {},
+   "source": [
+    "# Using the `BigInteger` to compile utility-scale quantum algorithms\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: I'm not sure about the title.
Maybe:
"Compiling Shor's algorithm at 2048 bit"
"Using BigInteger to compile Shor's algorithm at 2048 bit"

What do you think @positr0nium ?

File: documentation/source/general/tutorial/BigInteger.ipynb [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+    "\n",
+    "In a previous tutorial, we learned how Jasp allows us to use Qrisp to compile quantum algorithms.\n",
```
Comment (renezander90) [inline documentation/source/general/tutorial/BigInteger.ipynb]: at utility-scale

**Comment by positr0nium** [comment]
Don't merge yet. I'll run this notebook over night to get the results for 2048 bit :)
